src/gui/protocolwidget.cpp

```diff
@@ -104,71 +104,111 @@ QMenu *ProtocolWidget::showFilterMenu(QWidget *parent, Models::SignalledQSortFil
+            // Double-check if the file still exists
             if (QFileInfo::exists(localPath)) {
-                // keep in sync with ActivityWidget::slotItemContextMenu
-                menu->addAction(CommonStrings::showInFileBrowser(), parent, [localPath] {
-                    if (QFileInfo::exists(localPath)) {
-                        showInFileManager(localPath);
-                    }
-                });
-            }
-            // "Open in Browser" action
-            {
-                fetchPrivateLinkUrl(data.folder()->accountState()->account(), data.folder()->webDavUrl(), data.folder()->remotePathTrailingSlash() + data.path(), parent, [parent, menu = QPointer<QMenu>(menu)](const QUrl &url) {
-                    // as fetchPrivateLinkUrl is async we need to check the menu still exists
-                    if (menu) {
-                        menu->addAction(CommonStrings::showInWebBrowser(), parent, [url, parent] {
-                            Utility::openBrowser(url, parent);
-                        });
-                    }
-                });
+                showInFileManager(localPath);
             }
-            {
-                switch (data.status()) {
-                case SyncFileItem::DetailError:
-                    Q_FALLTHROUGH();
-                case SyncFileItem::SoftError:
-                    Q_FALLTHROUGH();
-                case SyncFileItem::BlacklistedError:
-                    menu->addAction(tr("Retry sync"), parent, [data, folder = QPointer<Folder>(data.folder())] {
-                        if (folder) {
-                            folder->journalDb()->wipeErrorBlacklistEntry(data.path());
-                            FolderMan::instance()->scheduler()->enqueueFolder(folder, SyncScheduler::Priority::Medium);
-                        }
-                    });
-                default:
-                    break;
+        });
+        if (!QFileInfo::exists(localPath)) {
+            showInFileBrowserAction->setEnabled(false);
+        }
+
+        // "Open in Browser" action
+        auto showInWebBrowserAction = menu->addAction(CommonStrings::showInWebBrowser());
+        showInWebBrowserAction->setEnabled(false);
+        fetchPrivateLinkUrl(data.folder()->accountState()->account(), data.folder()->webDavUrl(), data.folder()->remotePathTrailingSlash() + data.path(),
+            parent, [showInWebBrowserAction, parent, pos = menu->actions().size(), menu = QPointer<QMenu>(menu)](const QUrl &url) {
+                // as fetchPrivateLinkUrl is async we need to check the menu still exists
+                if (menu) {
+                    auto replacement = new QAction(CommonStrings::showInWebBrowser());
+                    connect(replacement, &QAction::triggered, [url, parent] { Utility::openBrowser(url, parent); });
+                    menu->insertAction(showInWebBrowserAction, replacement);
+                    menu->removeAction(showInWebBrowserAction);
```

Comment:
I think we should be able to update showInWebBrowserAction ?

Reply:
done